src/coreclr/vm/arm/thunktemplates.S

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "unixasmmacros.inc"
+#include "asmconstants.h"
+
+.syntax unified
+.thumb
+
+    .align 4
+
+PAGE_SIZE = 4096
+
+#define DATA_SLOT(stub, field) stub##Code + PAGE_SIZE + stub##Data__##field
+
+    LEAF_ENTRY StubPrecodeCode
+        ldr r12, DATA_SLOT(StubPrecode, MethodDesc)
+        ldr pc, DATA_SLOT(StubPrecode, Target)
```

Comment:
I wonder if we can use a single `ldp` here to load both registers in a single instruction

Reply:
We cannot, ldp requires consecutive register pair.

Reply:
Ah, sorry, ldp doesn't even exist on arm32. You probably meant ldm, but that doesn't support PC relative addressing